datastream-server-api/src/test/java/com/linkedin/datastream/server/TestFlushlessEventProducerHandler.java

```diff
@@ -157,9 +211,116 @@ public void testOutOfOrderAck() {
   }
 
   @Test
-  public void testBackwardsOrderAck() {
+  public void testOutOfOrderAckForNonComparableOffsets() {
```

Comment:
Technically, since you are expecting same results whether offsets are comparable or not, then you should not write these tests twice. Instead extract the method and have 2 tests and pass whether offsets are comparable or not.